File: lib/features/payments/data/models/payment_model.dart
final class PaymentModel {
  const PaymentModel({
    required this.id,
    required this.workerId,
    required this.amount,
    required this.paymentDate,
    required this.createdAt,
    required this.updatedAt,
  });

  final String id;
  final String workerId;
  final double amount;
  final DateTime paymentDate;
  final DateTime createdAt;
  final DateTime updatedAt;

  factory PaymentModel.fromJson(Map<String, dynamic> json) {
    return PaymentModel(
      id: json['id'] as String,
      workerId: json['workerId'] as String,
      amount: (json['amount'] as num).toDouble(),
      paymentDate: DateTime.parse(json['paymentDate'] as String),
      createdAt: DateTime.parse(json['createdAt'] as String),
      updatedAt: DateTime.parse(json['updatedAt'] as String),
    );
  }
}

File: Backend/src/modules/payments/models/payment.model.ts
import { Schema, model, Types, type Document, type Model } from 'mongoose';

export interface IPayment {
  workerId: Types.ObjectId;
  amount: number;
  paymentDate: Date;
}

export interface IPaymentDocument extends IPayment, Document {
  createdAt: Date;
  updatedAt: Date;
}

const paymentSchema = new Schema<IPaymentDocument>(
  {
    workerId: {
      type: Schema.Types.ObjectId,
      ref: 'Worker',
      required: true,
      index: true,
    },
    amount: {
      type: Number,
      required: true,
      min: 0,
    },
    paymentDate: {
      type: Date,
      required: true,
      default: Date.now,
      index: true,
    },
  },
  {
    timestamps: true,
    versionKey: false,
  },
);

paymentSchema.index({ workerId: 1, paymentDate: -1 });

export const PaymentModel: Model<IPaymentDocument> = model<IPaymentDocument>('Payment', paymentSchema);
